test/files/neg/t10806.scala

```diff
@@ -0,0 +1,26 @@
+/* scalac: -Xfatal-warnings
+ */
+
+trait T {
+
+  object Nope extends Throwable
+
+  def f(): Unit = {
+    // anything but simple type tests forced analysis
+    try { 1 } catch {
+      case _: Exception ⇒ println("Something went wrong")
+      case e: IllegalArgumentException ⇒ println(e.getMessage)
+      case Nope ⇒ ???
+    }
+
+    try { 1 } catch {
+      case _: Exception ⇒ println("Something went wrong")
+      case e: IllegalArgumentException ⇒ println(e.getMessage)
+    }
+
+    (new IllegalArgumentException()) match {
+      case _: Exception ⇒ println("Something went very wrong")
```

Comment:
is matching on an exception which you know to be there _really_ worse than catching an exception that should never happen?